### PR TITLE
Add users routes

### PR DESCRIPTION
**Pull request by dnewh**
This adds the remaining users routes to the Biome module:

REQUIRES `biome-credentials` FEATURE ENABLED FOR FULL USE

- `GET /biome/users` -- lists ID/username pairs for all users in the database
- `GET /biome/users/{id}` -- returns ID/username pair for the id in the request
- `DELETE /biome/users/{id}` -- removes a user and all their credentials from the database
    - takes a JSON payload consisting of:
        - username: username for the id provided in the request
        - hashed_password: hashed password for the id provided in the request
- `PUT /biome/users/{id}` -- updates a user's password in the database
    - takes a JSON payload consisting of:
       -username: username for the id provided in the request
       - hashed_password: hashed password for the id provided in the request
       - new_password: new hashed password

### CODE REVIEW

**Comment by dnewh** [comment]
This PR depends on changes introduced in PR #394

File: libsplinter/src/biome/credentials/credentials_store.rs [hunk 1]
```diff
@@ -58,22 +65,77 @@ impl CredentialsStore<UserCredentials> for SplinterCredentialsStore {
...
     ) -> Result<(), CredentialsStoreError> {
-        unimplemented!()
+        let credentials_builder: UserCredentialsBuilder = Default::default();
+        let credentials = credentials_builder
+            .with_user_id(user_id)
+            .with_username(username)
+            .with_password(password)
+            .build()
+            .map_err(|err| UserCredentialsBuilderError::BuildError(Box::new(err)))
+            .unwrap();
```
Comment (vaporos) [inline libsplinter/src/biome/credentials/credentials_store.rs]: unwrap() should be removed (turned into a ? in this case).

File: libsplinter/src/biome/credentials/store/mod.rs [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    ///  * `username` - The username the user uses for login
+    ///
+    fn fetch_credential_by_username(&self, username: &str) -> Result<T, CredentialsStoreError>;
+
+    fn fetch_username_by_id(&self, user_id: &str) -> Result<UsernameId, CredentialsStoreError>;
+
+    fn get_usernames(&self) -> Result<Vec<UsernameId>, CredentialsStoreError>;
+}
+
+impl From<UserCredentialsModel> for UserCredentials {
```
Comment (peterschwarz) [inline libsplinter/src/biome/credentials/store/mod.rs]: This should also be in the `diesel/mod.rs`, as the on below

File: libsplinter/src/biome/credentials/store/mod.rs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+/// Cost to encrypt password. The recommneded value is HIGH. Values LOW and MEDIUM may be used for
+/// development and testing as hashing and verifying passwords will be completed faster.
+#[derive(Debug, Deserialize, Clone)]
+pub enum PasswordEncryptionCost {
+    High,
+    Medium,
+    Low,
+}
+
+impl TryFrom<&str> for PasswordEncryptionCost {
```
Comment (peterschwarz) [inline libsplinter/src/biome/credentials/store/mod.rs]: You should use `FromStr`, which then allows you to do 
```rust
let cost: PasswordEncryptionCost = "high".parse()?;
```

File: libsplinter/src/biome/credentials/store/mod.rs [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+/// credentials without defining a storage strategy
+pub trait CredentialsStore<T> {
+    /// Adds a credential to the underlying storage
+    ///
+    /// # Arguments
+    ///
+    ///  * `credentials` - Credentials to be added
+    ///
+    ///
+    fn add_credentials(&self, credentials: UserCredentials) -> Result<(), CredentialsStoreError>;
```
Comment (peterschwarz) [inline libsplinter/src/biome/credentials/store/mod.rs]: Should the type of credentials here be `T`?

**Comment by dnewh** [comment]
@peterschwarz @rbuysse All comments have been addressed

**Pull request merged**